xwiki-platform-core/xwiki-platform-web/src/main/webapp/templates/passwd.vm

```diff
@@ -51,32 +42,44 @@ document.observe('xwiki:dom:loaded', function() {
           $doc.save("$services.localization.render('platform.core.profile.passwd.passwordChanged')", true)
         #end
       #end
-      #if($request.password)
-        #if($isValidPassword)
-          <span class='box infomessage'>$services.localization.render('platform.core.profile.passwd.success')</span>
-          <a href = "$doc.getURL("view")">$services.localization.render('platform.core.profile.passwd.return')</a>
-        #else
-          <span class='box errormessage'>$errorMessage</span>
-        #end
+      #if($request.password && $isValidPassword)
+        <span class='box infomessage'>$services.localization.render('platform.core.profile.passwd.success')</span>
+        <a href = "$doc.getURL("view")">$services.localization.render('platform.core.profile.passwd.return')</a>
       #end
       #if(!$isValidPassword)
-        #set($discard = $xwiki.jsfx.use('uicomponents/widgets/validation/livevalidation_prototype.js'))
-        #set($discard = $xwiki.ssfx.use('uicomponents/widgets/validation/livevalidation.css', true))
         $services.localization.render('platform.core.profile.passwd.instructions')
-
-      <form class="xform third" action="" method="post" autocomplete="off">
+        #set ($fields = [])
+        #set ($discard = $fields.add({
+          'name': 'originalpassword',
+          'label': $services.localization.render('platform.core.profile.passwd.originalPassword'),
+          'params': {
+            'type': 'password',
+            'size': '60',
+            'autocomplete': 'off'
+          },
+          'validate': {
+            'mandatory': {
+              'failureMessage': $services.localization.render('core.validation.required.message')
+            },
+            'programmaticValidation': {
+              'code': '$user.checkPassword($request.originalpassword)',
+              'failureMessage': $services.localization.render('platform.core.profile.passwd.invalidOriginalPassword')
+            },
+            'hideOkayMessage': 'true'
+          }
+        }))
+        #definePasswordFields($fields, 'password', 'password2')
+        ## Change the label for new password fields
+        #set ($fields[1].label = $services.localization.render('platform.core.profile.passwd.newPassword'))
+        ## This key fits better.
+        #set ($fields[2].label = $services.localization.render('xe.admin.passwordReset.step2.newPasswordVerification.label'))
+        <form class="xform third" action="" method="post" autocomplete="off">
         ## CSRF prevention
         <div class="hidden"><input type="hidden" name="form_token" value="$!{services.csrf.getToken()}" /></div>
-        <dl>
-          #if($isCurrentUsersProfile || !$hasAdmin)
```

Comment:
Just noticed this. Are you still allowing admins to change the password of a user without knowing the old password? AFACIT, you still consider when validating the password change, but I don't see where you consider this in the UI, to avoid asking the admin for the old password.

This is a usecase that was supported before and we must make sure we still support it.

Reply:
I'll check this.

Reply:
@Enygma2002, you're right, I am not considering that use case anymore. I'm reviewing the changes and come back with some commits.